beacon-chain/core/helpers/attestation.go

```diff
@@ -166,14 +166,29 @@ func ValidateAttestationTime(attSlot primitives.Slot, genesisTime time.Time, clo
 		lowerBoundsSlot,
 		currentSlot,
 	)
-	if attTime.Before(lowerBounds) {
-		attReceivedTooLateCount.Inc()
-		return errors.Join(ErrTooLate, attError)
-	}
 	if attTime.After(upperBounds) {
 		attReceivedTooEarlyCount.Inc()
 		return attError
 	}
+
+	attEpoch := slots.ToEpoch(attSlot)
+	if attEpoch < params.BeaconConfig().DenebForkEpoch {
+		if attTime.Before(lowerBounds) {
+			attReceivedTooLateCount.Inc()
+			return errors.Join(ErrTooLate, attError)
+		}
+		return nil
+	}
+
+	// The following checks are only for Deneb fork and beyond.
+
+	currentEpoch := slots.ToEpoch(currentSlot)
+	prevEpoch := currentEpoch.Sub(1)
```

Comment:
Not sure if anyone would start a testnet with at deneb, but this will panic if the current epoch is 0. Consider handling with SafeSub.

```suggestion
	prevEpoch, err := currentEpoch.SafeSub(1)
	if err != nil {
	   log.WithError(err).Debug("Ignoring underflow for a deneb attestation inclusion check in epoch 0")
	   prevEpoch = 0
	}
```